### ISSUE TITLE
memory leak on std.Thread.detach() on aarch64

### ISSUE THREAD

**Issue by cfal** [ziglang/zig#20677]
### Zig Version

0.13.0

std.mem.page_size: 4096
std.os.linux.tls.tls_image.alloc_align: 8
std.os.linux.tls.tls_image.alloc_size: 56

### Steps to Reproduce and Observed Behavior

```
const std = @import("std");

pub fn main() !void {
    while (true) {
        const conn_thread = try std.Thread.spawn(.{}, run_thread, .{});
        conn_thread.detach();
        std.time.sleep(1_000_000 * 1_000 * 1);
    }
}

fn run_thread() !void {
}
```

on aarch64 linux, the virtual memory seems to grow non-stop in increments of 16k or 32k, eg:

```
user     2416587  152  1.1 790980560 386048 pts/3 Rl+ 05:23   0:07 /home/user/.cache/zig/o/4da0e96851fc0b3a07b0dfc9f88a2d84/main
user     2416587  154  1.3 901823264 439936 pts/3 Rl+ 05:23   0:08 /home/user/.cache/zig/o/4da0e96851fc0b3a07b0dfc9f88a2d84/main
user     2416587  155  1.5 1011207080 493696 pts/3 Rl+ 05:23   0:09 /home/user/.cache/zig/o/4da0e96851fc0b3a07b0dfc9f88a2d84/main
user     2416587  158  4.8 3291481808 1606528 pts/3 Rl+ 05:23   0:30 /home/user/.cache/zig/o/4da0e96851fc0b3a07b0dfc9f88a2d84/main
user     2416587  158  5.6 3769915112 1840000 pts/3 Rl+ 05:23   0:34 /home/user/.cache/zig/o/4da0e96851fc0b3a07b0dfc9f88a2d84/main
```

while memory usage is constant on x86_64 or if doing `std.Thread.join()` instead. please let me know if this is not a zig problem.

### Expected Behavior

memory usage is constant

**Comment by alexrp** [comment]
@cfal in light of #20857, would you mind trying this on a recent build of master?

**Comment by alexrp** [comment]
I can repro on an RPi 5.

**Comment by alexrp** [comment]
I believe this is fixed by https://codeberg.org/ziglang/zig/pulls/30675.
